oz/__init__.py

```diff
@@ -131,7 +130,23 @@ def get_template_namespace(self):
     def initialize(self, **kwargs):
         self.trigger("initialize", **kwargs)
 
+    def set_secure_cookie(self, name, value, **kwargs):
+        print(name, value)
+        if self.settings["use_secure_cookie"]:
+            if "secure" not in kwargs:
+                kwargs["secure"] = True
+            if "httponly" not in kwargs:
+                kwargs["httponly"] = True
+        super(RequestHandler, self).set_secure_cookie(name, value, **kwargs)
+
     def prepare(self):
+        if self.settings["use_hsts"]:
+            self.set_header("Strict-Transport-Security", "max-age=2592000; includeSubDomains")
+            # Redirect to HTTPS if necessary
+            if self.request.headers.get("X-Forwarded-Proto", "https") != "https":
```

Comment:
This header is set by certain proxies, including heroku. But it won't always work, including if you directly visit the site. If this header isn't set, it should look at the request protocol (I think it's `self.request.proto`.)